Dip/Dip/Dip.swift

```diff
@@ -340,14 +385,43 @@ extension DependencyContainer.Tag: StringLiteralConvertible {
   }
 }
 
+/*
+.String("a") == .Stirng("a")
+.Int(1) == .Int(1)
+.String("1") == .Int(1)
+*/
 public func ==(lhs: DependencyContainer.Tag, rhs: DependencyContainer.Tag) -> Bool {
   switch (lhs, rhs) {
   case let (.String(lhsString), .String(rhsString)):
     return lhsString == rhsString
   case let (.Int(lhsInt), .Int(rhsInt)):
     return lhsInt == rhsInt
-  default:
-    return false
+  case let (.String(lhsString), .Int(rhsInt)):
+    return lhsString == String(rhsInt)
+  case let (.Int(lhsInt), .String(rhsString)):
+    return String(lhsInt) == rhsString
+  }
+}
+
+/**
+ Tags are compared by comparing strings created from their assiciated values.
+ 
+ .String("a") < .String("b")
+ .Int(0) < .Int(1)
+ .String("0") < .Int(1)
+ .Int(0) < .String("1")
+ 
+ */
+public func <(lhs: DependencyContainer.Tag, rhs: DependencyContainer.Tag) -> Bool {
```

Comment:
You are right, looks like it leaked from some WIP for other feature. I will remove that.